tutorials/general/t8_step7_interpolation.cxx

```diff
@@ -1,421 +1,415 @@
+   * This array has one entry per local element. */
+  double             *element_data = T8_ALLOC (double, num_elements);
+  /* The number of user defined data fields to write. */
+  int                 num_data = 1;
+  t8_vtk_data_field_t vtk_data;
+  vtk_data.type = T8_VTK_SCALAR;
+  strcpy (vtk_data.description, "Element own data");
+  vtk_data.data = element_data;
+  /* Copy the elment's data from the data array to the output array. */
+  for (ielem = 0; ielem < num_elements; ++ielem) {
+    element_data[ielem] = t8_element_get_value (data, ielem).values;
+  }
+
+  /* To write user defined data, we need to extended output function t8_forest_vtk_write_file
+   * from t8_forest_vtk.h. Despite writin user data, it also offers more control over which 
+   * properties of the forest to write. */
+  int                 write_treeid = 1;
+  int                 write_mpirank = 1;
+  int                 write_level = 1;
+  int                 write_element_id = 1;
+  int                 write_ghosts = 0;
+  t8_forest_write_vtk_ext (forest, prefix, write_treeid, write_mpirank,
+                           write_level, write_element_id, write_ghosts,
+                           0, 0, num_data, &vtk_data);
+  T8_FREE (element_data);
+}
+
+/* In this function the interpolation is described. In a first step a 
+ * hypercubic cmesh and then a forest is created. 
+ * We create a data array with the distance to the centroid of each cell.
+ * The forest is adapted and the data array is interpolated corresponding to
+ * the adapted forest.
+ * We write the uniform and the adapted forest to a vtu file.
+ */
+void
+t8_interpolation ()
+{
+  int                 level = 4;
+  t8_forest_t         forest_adapt;
+  t8_step7_element_data_t *elem_data;
+  t8_step7_adapt_data *data;
+  double              centroid[3];
+  const double        midpoint[3] = { 0.5, 0.5, 1 };
+  t8_scheme_cxx_t    *scheme = t8_scheme_new_default_cxx ();
+
+  /* Construct a cmesh */
+  t8_cmesh_t          cmesh =
+    t8_cmesh_new_hypercube (T8_ECLASS_PYRAMID, sc_MPI_COMM_WORLD, 0, 0, 0);
+
+  /* Construct a forest with one tree */
+  t8_forest_t         forest =
+    t8_forest_new_uniform (cmesh, scheme, level, 0, sc_MPI_COMM_WORLD);
+
+  /* Build initial data array and set data for the local elements. */
+  data = T8_ALLOC (t8_step7_adapt_data, 1);
+  elem_data = T8_ALLOC (t8_step7_element_data_t, 1);
+  data->element_data =
+    sc_array_new_count (sizeof (t8_step7_element_data_t),
+                        t8_forest_get_local_num_elements (forest));
+
+  const t8_locidx_t   numTrees = t8_forest_get_num_local_trees (forest);
+  /* Loop over all trees. The index of the data array is independent of the tree
+   * index. Thus, we set the index of the tree index to zero and add one in each 
+   * loop step of the inner loop.
+   */
+  int                 itree;
+  int                 ielem;
+  for (itree = 0, ielem = 0; itree < numTrees; itree++) {
```

Comment:
It does not work. Tried it before I commited it (with int instead of t8_locidx_t) - error: 
   expected ‘;’ before ‘ielem’
   ‘ielem’ was not declared in this scope
    expected ‘)’ before ‘;’ token
    ‘itree’ was not declared in this scope